src/main/java/com/adobe/commerce/cif/api/CustomerApi.java

```diff
@@ -64,7 +65,36 @@ Customer getCustomerById(
         @ApiParam(value = ACCEPT_LANGUAGE_DESC)
         @HeaderParam(ACCEPT_LANGUAGE) String acceptLanguage
     );
-    
+
+    @POST
+    @Path("/auth")
+    @ApiOperation(
+        nickname = "postAuthentication",
+        value = "Performs a authentication request to obtain a customer access token. Currently two types are " +
+            "supported: \"guest\" returns a token for a new guest (non-authenticated) customer and " +
+            "\"credentials\" authenticates a registered customer and returns a token for a successfully " +
+            "authentication otherwise results in an error"
+    )
+    @ApiResponses(value = {
+        @ApiResponse(code = HTTP_OK, message = HTTP_OK_MESSAGE, response = AuthenticationResponse.class),
+        @ApiResponse(code = HTTP_BAD_REQUEST, message = HTTP_BAD_REQUEST_MESSAGE, response = ErrorResponse.class),
+        @ApiResponse(code = HTTP_UNAUTHORIZED, message = HTTP_UNAUTHORIZED_MESSAGE, response = ErrorResponse.class)
+    })
+    @Consumes(MediaType.APPLICATION_JSON)
+    AuthenticationResponse authentication(
+        @ApiParam(value = "The type of authentication request.", required = true)
+        @FormParam("type")
```

Comment:
If you want to post a JSON body, these have to be `@ApiParam`, see for example `CartApi.postBillingAddress`. Here we use a wrapper so the `args` param in the action will have a address object like
```
other-stuff,
address {

},
more-stuff
```
but I'm not sure we want one here, at least from what we agreed.

Reply:
Changed to use `@ApiParam`, but I did not introduced a body wrapper object. The post body should be ```{
  "type" : "credentials",
  "username" : "user",
  "password" : "1234abcd"
}``` only.